Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/async/AsyncMessageProcessHelper.java

```diff
@@ -547,13 +569,11 @@ private String getPatientDiscoveryMessageCommunityId(
 
     protected Session getSession() {
         Session session = null;
-        SessionFactory fact = HibernateUtil.getSessionFactory();
-        if (fact != null) {
-            session = fact.openSession();
+        if (HibernateUtilFactory.getAsyncMsgsHibernateUtil() != null) {
```

Comment:
HibernateUtilFactory.getAsyncMsgsHibernateUtil() call twice and it impact performance since we look up the bean again applicationcontext.  Recommend to assign variable if sonar doesn't mark violation.